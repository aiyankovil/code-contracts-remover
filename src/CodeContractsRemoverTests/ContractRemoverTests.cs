using System;
using System.IO;
using System.Linq;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using CodeContractsRemover;
using CodeContractsRemover.CS;
using NUnit.Framework;

namespace CodeContractsRemoverTests
{
	[UseReporter(typeof(DiffReporter))]
	public class ContractRemoverTests
    {
	    private static string InitialFileName => TestsHelper.GetInitialFileName(".\\TestCases\\TestSubject.cs");

	    [Test]
        public void Process_ConvertMode_ResultIsCorrect()
        {
			//Arrange
			var file = $"{nameof(Process_ConvertMode_ResultIsCorrect)}.received.cs";
			File.Copy(InitialFileName, file);

			//Act
			ContractRemover.Process(file, ContractReplacementMode.Convert, Encoding.UTF8);

			//Assert
			Approvals.VerifyFile(file);
        }

        [Test]
        public void Process_FileWithLeadingComments_LeadingCommentsArePreserved()
        {
	        //Arrange
	        var file = $"{nameof(Process_FileWithLeadingComments_LeadingCommentsArePreserved)}.received.cs";
	        File.Copy(TestsHelper.GetInitialFileName(".\\TestCases\\FileWithLeadingComments.cs"), file);

	        //Act
	        ContractRemover.Process(file, ContractReplacementMode.Convert, Encoding.UTF8);

	        //Assert
	        Approvals.VerifyFile(file);
        }

        [Test]
        public void Process_ExtensionsCase_ResultIsCorrect()
        {
	        //Arrange
	        var file = $"{nameof(Process_ExtensionsCase_ResultIsCorrect)}.received.cs";
	        File.Copy(TestsHelper.GetInitialFileName(".\\TestCases\\ExtensionsCase.cs"), file);

	        //Act
	        ContractRemover.Process(file, ContractReplacementMode.ConvertAndAddAnnotations, Encoding.UTF8);

	        //Assert
	        Approvals.VerifyFile(file);
        }

		[Test]
        public void Process_NothingToModify_FileNotModified()
        {
	        //Arrange
	        var file = $"{nameof(Process_NothingToModify_FileNotModified)}.received.cs";
	        File.Copy(TestsHelper.GetInitialFileName(".\\TestCases\\NothingToModify.cs"), file);

	        //Act
	        var changed = ContractRemover.Process(file, ContractReplacementMode.Convert, Encoding.UTF8);

	        //Assert
	        Assert.False(changed);
        }

		[Test]
        public void Process_StatisticMode_ResultIsCorrect()
        {
	        //Arrange
	        var file = $"{nameof(Process_StatisticMode_ResultIsCorrect)}.received.cs";
	        File.Copy(InitialFileName, file);

	        //Act
	        ContractRemover.Process(file, ContractReplacementMode.Stats, Encoding.UTF8);

	        //Assert
	        Approvals.Verify(CSharpStatsCollector.GetStats());
        }
    }
}
