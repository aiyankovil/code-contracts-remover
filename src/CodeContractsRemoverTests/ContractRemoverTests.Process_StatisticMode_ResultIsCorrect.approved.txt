[ContractClassFor]                                	  1
[ContractInvariantMethod]                         	  2
Contract.Assert                                   	  1
Contract.Ensures                                  	  6
Contract.Invariant                                	  2
Contract.Requires                                 	  6
Contract.Result                                   	  5
